packages/source-gdocs-paste/src/index.ts

```diff
@@ -1,35 +1,40 @@
-import { Annotation } from '@atjson/document';
+import Document, { Annotation } from '@atjson/document';
+import schema from '@atjson/schema';
+import gdocsSchema from './schema';
 
-import extractTextStyles from './text-styles';
-import extractParagraphStyles from './paragraph-styles';
-import extractListStyles from './list-styles';
-
-export default class Parser {
-
-  gdocsSource: string;
+import GDocsParser from './gdocs-parser';
 
+export default class extends Document {
   constructor(gdocsSource: string) {
-    this.gdocsSource = gdocsSource;
-  }
 
-  getContent(): string {
-    return this.gdocsSource.resolved.dsl_spacers;
+    let gdocsParser = new GDocsParser(gdocsSource);
+
+    super({
+      content: gdocsParser.getContent(),
+      contentType: 'text/google-docs',
+      annotations: gdocsParser.getAnnotations(),
+      schema: gdocsSchema
+    });
   }
 
-  getAnnotations(): Annotation[] {
-    let transforms = {
-      'text': extractTextStyles,
-      'paragraph': extractParagraphStyles,
-      'list': extractListStyles
-    }
-
-    let annotations = this.gdocsSource.resolved.dsl_styleslices.map(styleSlice => {
-      if (transforms[styleSlice.stsl_type]) {
-        return transforms[styleSlice.stsl_type](styleSlice.stsl_styles, this.gdocsSource.resolved.dsl_entitymap);
-      }
+  toCommonSchema(): Document {
+    let doc = new Document({
+      content: this.content,
+      contentType: 'text/atjson',
+      annotations: [...this.annotations],
+      schema
     });
 
-    return [].concat.apply([], annotations).filter(a => a !== undefined);
-  }
+    doc.where({ type: '-gdocs-ts_bd' }).set({ type: 'bold' });
+    doc.where({ type: '-gdocs-ts_it' }).set({ type: 'italic' });
 
+    doc.where({ type: '-gdocs-ps_hd' }).set({ type: 'heading' });
+    doc.where({ type: '-gdocs-ps_hd' }).map({ attributes: { '-gdocs-level': 'level' });
```

Comment:
This won't actually work, since all the types of `-gdocs-ps_hd` have now been renamed to `heading`. If you chain the call to `map` after the `set`, this should work.